lib/mpl_toolkits/mplot3d/proj3d.py

```diff
@@ -218,69 +189,12 @@ def proj_trans_clip_points(points, M):
     xs, ys, zs = list(zip(*points))
     return proj_transform_clip(xs, ys, zs, M)
 
-def test_proj_draw_axes(M, s=1):
-    import pylab
-    xs, ys, zs = [0, s, 0, 0], [0, 0, s, 0], [0, 0, 0, s]
-    txs, tys, tzs = proj_transform(xs, ys, zs, M)
-    o, ax, ay, az = (txs[0], tys[0]), (txs[1], tys[1]), \
-            (txs[2], tys[2]), (txs[3], tys[3])
-    lines = [(o, ax), (o, ay), (o, az)]
-
-    ax = pylab.gca()
-    linec = LineCollection(lines)
-    ax.add_collection(linec)
-    for x, y, t in zip(txs, tys, ['o', 'x', 'y', 'z']):
-        pylab.text(x, y, t)
-
-def test_proj_make_M(E=None):
-    # eye point
-    E = E or np.array([1, -1, 2]) * 1000
-    #E = np.array([20,10,20])
-    R = np.array([1, 1, 1]) * 100
-    V = np.array([0, 0, 1])
-    viewM = view_transformation(E, R, V)
-    perspM = persp_transformation(100, -100)
-    M = np.dot(perspM, viewM)
-    return M
-
-def test_proj():
-    import pylab
-    M = test_proj_make_M()
-
-    ts = ['%d' % i for i in [0,1,2,3,0,4,5,6,7,4]]
-    xs, ys, zs = [0,1,1,0,0, 0,1,1,0,0], [0,0,1,1,0, 0,0,1,1,0], \
-            [0,0,0,0,0, 1,1,1,1,1]
-    xs, ys, zs = [np.array(v)*300 for v in (xs, ys, zs)]
-    #
-    test_proj_draw_axes(M, s=400)
-    txs, tys, tzs = proj_transform(xs, ys, zs, M)
-    ixs, iys, izs = inv_transform(txs, tys, tzs, M)
-
-    pylab.scatter(txs, tys, c=tzs)
-    pylab.plot(txs, tys, c='r')
-    for x, y, t in zip(txs, tys, ts):
-        pylab.text(x, y, t)
-
-    pylab.xlim(-0.2, 0.2)
-    pylab.ylim(-0.2, 0.2)
-
-    pylab.show()
 
 def rot_x(V, alpha):
```

Comment:
This function is not used internally, exists since 2009 with en error and no one noticed that. Maybe it is better to remove it?
P.S. I even do not understand what is the first coordinate of the vector (?; X, Y, Z). Why it is 4D and what is the used for?